### PR TITLE
Bugfix/click component

### PR DESCRIPTION
**Pull request by justinehell**
We found a bug between the click and drag'n'drop of a component -> they have a different behaviour due to the different `path` given in the `addComponent` fonction. So we decided to do a refactorization to clean the code and have a less complex management of the path of a component.

Refacto : add a component with the full path, instead of separated `folder` and `fileName`. 

:information_source:  Don't forget to modify the changelog.md before merging this branch.

### CODE REVIEW

File: src/models/DefaultData.js [hunk 1]
```diff
@@ -98,6 +98,7 @@ class DefaultData {
      * Plugin configuration storage.
      * @type {DefaultConfiguration}
      */
+    // MAYBE NOT USEFULL ANYMORE ?
```
Comment (k-y0u) [inline src/models/DefaultData.js:101]: to remove ?

File: guides/migrations/0.20.0_to_0.x.0.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Migrate from version 0.20.0 to 0.x.0
```
Comment (k-y0u) [inline guides/migrations/0.20.0_to_0.x.0.md:1]: not in phase with changelog

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/models/DefaultData.js (hunk 1 above)
Comment (justinehell) [reply src/models/DefaultData.js]: removed

**Review by k-y0u — DISMISSED** [review]

File: src/models/DefaultConfiguration.js [hunk 3]
```diff
@@ -20,6 +20,7 @@ class DefaultConfiguration {
    * components & links.
    * @param {number} [props.singleComponentParams.precision] - Space interval between coordinates
    * that will be tested. Performance decreases proportionally to the square of this parameter.
+   * @param {boolean} [props.isFolderTypeDiagram] - True if diagram type is folder, otherise false.
```
Comment (Zorin95670) [inline src/models/DefaultConfiguration.js:23]: otherwise

File: guides/migrations/0.21.0_to_0.XX.0.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
+# Migrate from version 0.21.0 to 0.XX.0
+
+## Update in `DefaultPlugin`
+
+- Merge `folder` and `fileName` params in `addComponent` method:
```
Comment (Zorin95670) [inline guides/migrations/0.21.0_to_0.XX.0.md:5]: add concrete example to let people see the difference.

File: guides/migrations/0.21.0_to_0.XX.0.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    rootId,
+    definition,
+    path,
+    drawOption = null,
+  )
+```
+
+## Update in `DefaultData`
+
+- Merge `folder` and `fileName` params in `addComponent` method:
```
Comment (Zorin95670) [inline guides/migrations/0.21.0_to_0.XX.0.md:28]: same here

File: src/models/DefaultConfiguration.js (hunk 3 above)
Comment (justinehell) [reply src/models/DefaultConfiguration.js:23]: done

File: changelog.md [hunk 6]
```diff
@@ -5,6 +5,16 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html)
 
+## [Unreleased]
```
[the remark this answers is not in the document]
Comment (justinehell) [reply changelog.md:8]: done

File: guides/migrations/0.21.0_to_0.XX.0.md (hunk 5 above)
Comment (justinehell) [reply guides/migrations/0.21.0_to_0.XX.0.md:32]: done

File: guides/migrations/0.21.0_to_0.XX.0.md (hunk 4 above)
Comment (justinehell) [reply guides/migrations/0.21.0_to_0.XX.0.md:5]: done

**Review by Zorin95670 — APPROVED** [review]

**Review by k-y0u — APPROVED** [review]

**Comment by justinehell** [comment]
New branch name -> new PR https://github.com/ditrit/leto-modelizer-plugin-core/pull/237

**Pull request closed without merging**